src/core/common/string.hpp

```diff
@@ -258,6 +258,17 @@ template <uint16_t SIZE> class String : private StringBase
  *
  */
 
+/**
+ * This function validates whether a given byte sequence (string) follows UTF-8 encoding.
+ *
+ * @param[in]  aString  A null-terminated byte sequence.
+ *
+ * @retval TRUE   The sequence is a valid UTF-8 string.
+ * @retval FALSE  The sequence is not a valid UTF-8 string.
+ *
+ */
+bool ValidateUtf8(const char *aString);
```

Comment:
nit: 
```suggestion
bool IsValidUtf8String(const char *aString);
```

since it returns `bool`.

Reply:
Done.